src/test/java/org/apache/datasketches/memory/test/AllocateDirectWritableMapMemoryTest.java

```diff
@@ -45,6 +53,28 @@
 public class AllocateDirectWritableMapMemoryTest {
   private static final String LS = System.getProperty("line.separator");
 
+  static final Method IS_FILE_READ_ONLY;
+  static final Method GET_CURRENT_DIRECT_MEMORY_MAP_ALLOCATIONS;
+  
+  static {
+    IS_FILE_READ_ONLY =
+        ReflectUtil.getMethod(ReflectUtil.ALLOCATE_DIRECT_MAP, "isFileReadOnly", File.class);
+    GET_CURRENT_DIRECT_MEMORY_MAP_ALLOCATIONS =
+        ReflectUtil.getMethod(ReflectUtil.BASE_STATE, "getCurrentDirectMemoryMapAllocations", (Class<?>[])null); //static
+  }
+  
+  private static boolean isFileReadOnly(final File file) {
+    try {
+      return (boolean) IS_FILE_READ_ONLY.invoke(null, file);
+    } catch (Exception e) { throw new RuntimeException(e); }
+  }
+
+  private static long getCurrentDirectMemoryMapAllocations() {
+    try {
+      return (long) GET_CURRENT_DIRECT_MEMORY_MAP_ALLOCATIONS.invoke(null);
+    } catch (Exception e) { throw new RuntimeException(e); }
+  }
```

Comment:
The previous strategy imported static methods into the test suite - is the new strategy to redefine them locally and delegate to the actual static method using reflection?  This has minimal impact on the existing tests and is a nice approach.

Reply:
I have revised this even more.  Check the new code and my comment below.